bgpd/bgp_main.c

```diff
@@ -469,6 +476,23 @@ main (int argc, char **argv)
 	  if (vty_port <= 0 || vty_port > 0xffff)
 	    vty_port = BGP_VTY_PORT;
 	  break;
+	case OPTION_VTYSOCK:
+	  vty_sock_name = strrchr (BGP_VTYSH_PATH, '/');
```

Comment:
I think only the "case OPTION_VTYSOCK" section could be moved to a separate function, but didn't expect this to make sense as it's short and all the other parts would still have to be in each daemon. But ok to move if it requested (Not sure if this was a question, suggestion or request)